erpnext/hr/doctype/hr_settings/hr_settings.py

```diff
@@ -22,3 +24,12 @@ def validate_password_policy(self):
 		if self.email_salary_slip_to_employee and self.encrypt_salary_slips_in_emails:
 			if not self.password_policy:
 				frappe.throw(_("Password policy for Salary Slips is not set"))
+	
+	def on_update(self):
+		self.toggle_rounded_total()
+		frappe.clear_cache()
+
+	def toggle_rounded_total(self):
```

Comment:
This will only hide the rounded total field. If rounded total is disabled then net pay should not be rounded off and amount in words should also be equal to net pay in this case